Source/AliveLib/MainMenu.cpp

```diff
@@ -2062,8 +2064,15 @@ signed int MainMenuController::LoadDemo_Update_4D1040(DWORD)
         strcpy(&lvFilename[7], sPathData_559660.paths[levelId].field_22_lvl_name_cd);
         auto lvFilenameNoPrefix = &lvFilename[7];
 
-        while (!MainMenuController::checkIfDemoFileExists_4D1430(lvFilenameNoPrefix) && !MainMenuController::checkIfDemoFileExists_4D1430(lvFilename))
+#if BEHAVIOUR_CHANGE_SUB_DATA_FOLDERS
+		std::string subFolderPath(&lvFilenameNoPrefix[1]);
```

Comment:
All this stl usage will make PS2 ports and others a lot harder to do :(

Reply:
fair point. ill use some older techniques instead